OsmAnd-java/src/main/java/net/osmand/router/HHRoutePlanner.java

```diff
@@ -797,6 +797,12 @@ public int compare(HHRouteRegionsGroup<T> o1, HHRouteRegionsGroup<T> o2) {
 		if (allMatched) {
 			return currentCtx;
 		}
+		if (RoutePlannerFrontEnd.CALCULATE_MISSING_MAPS && groups.size() > 1) {
```

Comment:
We need filter always if groups > 1.  Why here is the check for CALCULATE_MISSING_MAPS